Fix compilation error when having getTemplate() property (issue #48)

dorireuv opened the pull request:
Fix issue #48: Compilation error when having getTemplate() property.

cjp39 commented on src/test/java/org/inferred/freebuilder/processor/ListMultimapPropertyFactoryTest.java:
```diff
@@ -628,11 +628,11 @@ public void testMergeFrom_builder() {
         .with(MULTIMAP_PROPERTY)
         .with(testBuilder()
-            .addLine("DataType.Builder template = DataType.builder()")
+            .addLine("DataType.Builder _template = DataType.builder()")
```
Could you revert all the changes to tests that are just using a variable named "template"?

cjp39 commented:
Thanks, Dori! It just needs a new test in ProcessorTest.java of a value type with a getTemplate() method, to ensure we don't regress in future.

cjp39 commented:
Also, don't forget to sign the [Google Individual Contributor License Agreement](https://developers.google.com/open-source/cla/individual?csw=1)

dorireuv commented:
Reverted the unrelated changes and added a test (verified that it fails before the fix).
I also signed the Google Individual Contributor License Agreement.

cjp39 commented:
Perfect; thanks!
